nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -725,6 +745,32 @@ func resourceNsxtEdgeTransportNodeCreate(d *schema.ResourceData, m interface{})
 	connector := getPolicyConnector(m)
 	client := nsx.NewTransportNodesClient(connector)
 
+	nodeID := d.Get("node_id").(string)
+	if nodeID != "" {
+		obj, err := client.Get(nodeID)
+		if err != nil {
+			return handleCreateError("TransportNode", nodeID, err)
+		}
+		// Set node_id, revision and computed values in schema
+		d.Set("failure_domain", obj.FailureDomainId)
+
+		converter := bindings.NewTypeConverter()
+		base, errs := converter.ConvertToGolang(obj.NodeDeploymentInfo, mpmodel.EdgeNodeBindingType())
+		if errs != nil {
+			return handleCreateError("TransportNode", nodeID, errs[0])
+		}
+		node := base.(mpmodel.EdgeNode)
+		d.Set("external_id", node.ExternalId)
+		d.Set("fqdn", node.Fqdn)
+		d.Set("ip_addresses", node.IpAddresses)
+		if obj.Revision != nil {
+			d.Set("revision", obj.Revision)
+		}
+
+		d.SetId(nodeID)
+		return resourceNsxtEdgeTransportNodeUpdate(d, m)
```

Comment:
In case update fails, there might be a state problem here. Since we already performed `d.SetId(nodeId)`, the resource will be already present in state, if I'm not mistaken. 
If would be safer to catch error here, and in case Update fails, nullify the Id.